### DISCUSSION TITLE
Disabling parsers for certain files

### DISCUSSION THREAD

**Discussion by KonnorRogers** [prettier/prettier#16945] [Q&A]
Hi there! I'm working on a project using Lit tagged template literals that look like this:

```js
render() { return html`` }
```

When I add the `prettier-plugin-jinja-template` it starts flagging these `.ts` files as well as flagging CSS files and other files.

I was curious if theres a way to make a prettier parser only run on certain files and not get added to the "global parser registry"

I tried doing the following:

```js
const config = {
  overrides: [
    // TODO: Need to find a way to disable jinja from running on .ts files.
    {
      files: ['./docs/**/*.njk'],
      options: {
        parser: 'jinja-template',
        plugins: ['prettier-plugin-jinja-template']
      },
    },
  ],
};

export default config;
```

But that didn't seem to work as the languages defined in the jinja plugin seem to be added anyways.

Is there a config option im missing here to make a parser / plugin only run on specific files?

**Comment by Poliklot** [comment]
Hi, @KonnorRogers!

I know this is an old question, but maybe this helps someone who finds this
discussion later while debugging a similar setup.

I tested this with a small local project using the versions that were current
around the time this was posted:

- `prettier@3.4.2`
- `prettier-plugin-jinja-template@2.0.0`

Putting the plugin inside an `overrides` entry worked for me. For `.ts` and
`.css` files, Prettier did not load the plugin. For the matching `.njk` file, it
loaded `parser: "jinja-template"` and the plugin.

This config worked:

```js
// prettier.config.mjs
export default {
  overrides: [
    {
      files: ["docs/**/*.njk"],
      options: {
        parser: "jinja-template",
        plugins: ["prettier-plugin-jinja-template"],
      },
    },
  ],
};
```

I checked it with:

```sh
prettier src/lit.ts --config prettier.config.mjs --log-level debug
prettier src/style.css --config prettier.config.mjs --log-level debug
prettier docs/page.njk --config prettier.config.mjs --log-level debug
```

For `src/lit.ts` and `src/style.css`, the loaded options did not include
`plugins`. For `docs/page.njk`, the loaded options included:

```json
{
  "parser": "jinja-template",
  "plugins": ["prettier-plugin-jinja-template"]
}
```

I also checked the API case in one Node process: format a Jinja string with the
plugin first, then format a TypeScript string without passing the plugin. The
TypeScript format still used the normal `typescript` parser, so the plugin did
not seem to stay in some global parser registry for the next call.

So I don't think there is a separate "disable this parser for other files"
option. The plugin just should not be loaded for those files in the first place.

If it still runs on `.ts` or `.css` files, I would check whether the plugin is
also being loaded somewhere else, for example:

- top-level `plugins` in the Prettier config
- a CLI flag like `--plugin=prettier-plugin-jinja-template`
- editor settings
- another Prettier config file higher up the directory tree
- a wrapper tool that always passes the plugin

Also, one detail: `prettier-plugin-jinja-template@2.0.0` registers these
extensions:

```js
[".jinja", ".jinja2", ".j2", ".html"]
```

It does not register `.njk`, so for `.njk` files the explicit override with
`parser: "jinja-template"` is needed.

If you want the most isolated setup, another safe option is to use two scripts:

```json
{
  "scripts": {
    "format": "npm run format:code && npm run format:jinja",
    "format:code": "prettier --write \"src/**/*.{ts,tsx,js,css,json,md}\"",
    "format:jinja": "prettier --write \"docs/**/*.njk\" --plugin=prettier-plugin-jinja-template --parser=jinja-template"
  }
}
```

But with the config above, the override-only setup worked in my test.
